lib/net/http_fetch_plugin.js

```diff
@@ -54,8 +63,38 @@ shaka.net.HttpFetchPlugin = class {
       timedOut: false,
     };
 
+    const connectionTimeoutMs = request.retryParameters.connectionTimeout;
+    if (connectionTimeoutMs) {
+      connectionTimer = new shaka.util.Timer(() => {
+        abortStatus.timedOut = true;
+        controller.abort();
+      });
+
+      connectionTimer.tickAfter(connectionTimeoutMs / 1000);
+    }
+
+    const stallTimeoutMs = request.retryParameters.stallTimeout;
+    if (stallTimeoutMs) {
+      stallTimer = new shaka.util.Timer(() => {
+        abortStatus.timedOut = true;
+        controller.abort();
+      });
+    }
+
+    /** @type {shaka.extern.ProgressUpdated} */
+    const onProgressUpdated = (time, bytes, numBytesRemaining) => {
+      if (connectionTimer) {
+        connectionTimer.stop();
+      }
+      if (stallTimer) {
+        stallTimer.stop();
```

Comment:
tickAfter implies a reset, so you can drop the stop() call here.

Reply:
Done!